net/fan_in_listener_factory.go

```diff
@@ -0,0 +1,147 @@
+package net
+
+import (
+	"github.com/sirupsen/logrus"
+
+	"net"
+)
+
+type fanInListenerFactory struct {
+	factories []ListenerFactory
+	logger    *logrus.Entry
+}
+
+type FanInListenerFactoryArgs struct {
+	Factories []ListenerFactory
+	Logger    *logrus.Logger
+}
+
+func NewFanInListenerFactory(args *FanInListenerFactoryArgs) (ListenerFactory, error) {
```

Comment:
Btw. the way you solved the creation of shared and separate listeners is pretty elegant 👌 

I needed to do 2 passes, but to get how it works, but it's nice! A few comments could help 😉

Reply:
Thanks! And yeah, I'll add in a few comments for clarification